### ISSUE TITLE
Re-using the Deep-net Model

### ISSUE THREAD

**Issue by PraveerSINGH** [imatge-upc/saliency-2016-cvpr#10]
Hey,

I was trying to re-use your deep-net caffe model in my own Torch network to compare saliency maps from two seperate inputs using KL Divergence loss. (Note that I do-not fine-tune your model while training my network; it is simply used for outputting saliency maps and backward propagating the loss through the gradients wrt its input). 

I am curious to know if its simply a plug and play model (apart from taking care of input characteristics of a caffe model like BGR format and input value ranging b/w 0 -> 255) or are there some other pre and post processing steps involved in utilising the model.

This is because I see that in the _init_.py file, you apply various pre-processing as well as post processing to the input image and to the Saliency Maps (as mentioned in your paper), but I am not clear if they are to be applied for test purposes as well ?

Thanks in advance for your response to this thread.

**Comment by kevinmcguinness** [comment]
The preprocessing steps are basically mean subtraction, transposition to caffe order, and scaling of the input values. These should be performed at test time too. 

The post-processing steps are only really needed to scale the outputs to [0, 255] to produce a saliency image. These can be omitted if you are just using the model to backprop wrt the input image. I guess you could implemented this scaling as another layer too if you need to. There's optional blurring of the saliency map in there too, but this can be also be omitted. None of the postprocessing steps were used when the model was being trained.
